symengine/cwrapper.h

```diff
@@ -107,6 +127,42 @@ int is_a_Rational(const basic s);
 //! Return 1 if s is an Symbol, 0 if not.
 int is_a_Symbol(const basic s);
 
+
+//! Wrapper for std::vector<int>
+
+typedef struct CVectorInt CVectorInt;
+
+CVectorInt* vectorint_new();
+
+// 'data' must point to allocated memory of size 'size'. The function returns 0
+// if std::vector<int> can be initialized using placement new into 'data',
+// otherwise 1 if 'size' is too small or 2 if 'data' is not properly aligned.
+// No memory is leaked either way. Use vectorint_placement_new_check() to check
+// that the 'data' and 'size' is properly allocated and aligned. Use
+// vectorint_placement_new() to do the actual allocation.
+int vectorint_placement_new_check(void *data, size_t size);
+CVectorInt* vectorint_placement_new(void *data);
+
+void vectorint_placement_free(CVectorInt *self);
+
+void vectorint_free(CVectorInt *self);
+void vectorint_push_back(CVectorInt *self, int value);
+int vectorint_get(CVectorInt *self, int n);
+
+//! Wrapper for vec_basic
+
+typedef struct CVecBasic CVecBasic;
+
+CVecBasic* vecbasic_new();
+void vecbasic_free(CVecBasic *self);
+void vecbasic_push_back(CVecBasic *self, const basic value);
+void vecbasic_get(CVecBasic *self, int n, basic result);
```

Comment:
Well, the other argument is that the `self` variable should always be first, since these are just methods in the given object. Is there some guideline or some other project which has return arguments first?